rpc/gnmi/mixed-schema.md

```diff
@@ -99,6 +96,35 @@ constrained to the specified origin. Particularly:
   the specified `origin`. To delete contents from multiple origins, a client
   MUST specify multiple paths within the `delete` of the `SetRequest`.
 
+### Special considerations for the use of Origin CLI
+
+The following rules apply to a `SetRequest` containing `origin` CLI.  TODO:
+Special considerations for mixing native schema are currently undefined.
+
+* If a `SetRequest` contains `origin` `cli` and `origin` `openconfig`, the CLI
+  MUST be given precedence over the OpenConfig origin.  This precedence is
+  independent of the order in which origin CLI appears.  This overrides the
+  gNMI specification requirement of ordered path based precedence within a
+  `SetRequest`.  The goal of this precedence is to create deterministic
+  behavior for resolving overlapping configuration.  For setRequests without
+  `origin` CLI, order based precedence still applies.
+* The `update` operation MUST be supported for setRequests containing `origin`
+  `cli` and `origin` `openconfig`.  The `update` operation is compatible with
+  the requirements for `origin` `cli` to take precedence.
+* The `replace` operation for a setRequest containing only `origin` `cli` MUST
+  be treated as a full device configuration replacement.
```

Comment:
Yes, this assumes that to be useful, the `replace` operator for `origin` `cli` should cover all functionality of the device.  Is this a rhetorical comment or does JunOS not support all functions through it's "set" and {} curly brace syntax?

Reply:
JUNOS does support this but I raise this abstract of the implementation I'm associated with because it was never the intention of OpenConfig modeling to actually be exposed in vendor's CLIs - those that have done so, likely have due to already existing architectural principles and byproducts but this is not the case for all implementations.

With this verbiage, I'm afraid that 3rd party schema coverage for vendor CLIs is then implicitly mandated all of which will have different user experiences/representations.  Those that do not support now need to in order to comply or we have diverged implementations and thus behaviors of a specification that cannot be conformed to.  Essentially this silently raises the barrier of entry.

I personally think mandating CLI behaviors is scope creep at this stage.  In addition there are already other behaviors occurring that are starting to exclude other NBI that have me seriously concerned.